core/chains/evm/config/toml/defaults.go

```diff
@@ -98,39 +117,60 @@ func init() {
 			continue
 		}
 
+		// read the file to bytes
 		path := evmDir + "/" + entry.Name()
-		file, err := os.Open(path)
+		chainID, chain, err := readConfig(path, os.ReadFile, entry.Name() == "fallback.toml")
 		if err != nil {
-			log.Fatalf("error opening file (name: %v) in custom defaults override directory: %v", entry.Name(), err)
+			if errors.Is(err, errFallbackConfig) {
```

Comment:
We already know when it is a fallback config, because we just passed the boolean arg to `readConfig`. Let's use that logic directly instead of hacking errors.